balancer/endpointsharding/endpointsharding.go

```diff
@@ -76,24 +75,28 @@ type endpointSharding struct {
 	mu sync.Mutex // Sync updateState callouts and childState recent state updates
 }
 
-// UpdateClientConnState creates a child for new endpoints and deletes children
-// for endpoints that are no longer present. It also updates all the children,
-// and sends a single synchronous update of the childrens' aggregated state at
-// the end of the UpdateClientConnState operation. If any endpoint has no
-// addresses, returns error without forwarding any updates. Otherwise returns
-// first error found from a child, but fully processes the new update.
-func (es *endpointSharding) UpdateClientConnState(state balancer.ClientConnState) error {
-	if len(state.ResolverState.Endpoints) == 0 {
+// ValidateEndpoints returns an error if the endpoints list is empty, or no
+// addresses are present in endpoint list.
+func ValidateEndpoints(endpoints []resolver.Endpoint) error {
+	if len(endpoints) == 0 {
 		return errors.New("endpoints list is empty")
 	}
```

Comment:
I think you can just delete this?  It's a subset of the other validation?

Reply:
I kept this here soley for the sake of the more specific error string. I think the two are distinct, which is why I left this clause for a different error message.